gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingChannelProvider.java

```diff
@@ -100,26 +86,42 @@ public boolean needsExecutor() {
   }
 
   @Override
-  public ManagedChannel getChannel() throws IOException {
-    if (needsExecutor()) {
-      throw new IllegalStateException("getChannel() called when needsExecutor() is true");
-    } else {
-      return createChannel(executorProvider.getExecutor());
-    }
+  public TransportChannelProvider withExecutor(ScheduledExecutorService executor) {
+    return toBuilder().setExecutorProvider(FixedExecutorProvider.create(executor)).build();
   }
 
   @Override
-  public ManagedChannel getChannel(Executor executor) throws IOException {
-    if (!needsExecutor()) {
-      throw new IllegalStateException("getChannel(Executor) called when needsExecutor() is false");
+  public boolean needsHeaders() {
```

Comment:
I guess it means "needs headers to be provided." If the `headerProvider` is not null, then we have what we need.

Both `needsExecutor()` and `needsHeaders()` have to be public because they tell the caller if they need to call `withExecutor` / `withHeaders` before calling `getTransportChannel()`.